rust/lance-encoding/src/encodings/physical/binary.rs

```diff
@@ -289,44 +302,89 @@ impl MiniBlockDecompressor for BinaryMiniBlockDecompressor {
 }
 
 /// Most basic encoding for variable-width data which does no compression at all
+/// The DataBlock memory layout looks like below:
+/// ----------------------------------------------------------------------------------------
+/// | bits_per_offset | number of values  | bytes_start_offset | offsets data | bytes data
+/// ----------------------------------------------------------------------------------------
+/// |       1 byte    | bits_per_offset/8 | bits_per_offset/8  |  offsets_len | dat_len
```

Comment:
Does the `number of values` need to change with the data type?  Can it always be 32 bits?  A single page will never have more than 2^32 values in it.